dv/uvm/core_ibex/Makefile

```diff
@@ -132,6 +124,25 @@ gen-dirs := $(OUT) $(OUT-SEED) $(metadata) $(OUT)/rtl_sim
 $(gen-dirs): %:
 	mkdir -p $@
 
+# sim-cfg-mk is a makefile fragment that sets-up anything simulator specific, it
+# is generated by sim_makefrag_gen.py
+sim-cfg-mk = $(OUT)/.sim-cfg.mk
+
+# The FORCE pre-requisite causes sim-cfg-mk to always be built, however this
+# also triggers a makefile restart. The MAKE_RESTARTS guard around this target
+# ensures this doesn't result in an infinite recursion.
```

Comment:
For the next person trying to get this working in their Makefile, maybe we should be a bit more helpful (it took us quite a few goes!). Maybe something like this?

```make
# The include of $(sim-cfg-mk) below tells Make that it should ensure the file
# exists. This rule tells Make how to build it. We also want to ensure it's
# built on every run. Declaring the rule to be .PHONY doesn't work because it
# causes Make to no longer think the rule actually generates the file. If you
# do that, the file gets re-built, but Make doesn't read the new contents. So
# instead we depend on FORCE (a phony target). This ensures a rebuild, but also
# causes an infinite recursion: when we re-read this file to include the new
# contents of $(sim-cfg-mk), we run the rule again. To avoid that, we check for
# MAKE_RESTARTS, which is defined on re-runs. Phew!
```